core/modules/menu/menu.admin.inc

```diff
@@ -446,29 +484,43 @@ function menu_edit_item($form, &$form_state, $type, $item, $menu) {
     '#delta' => _menu_get_menu_weight_delta($menu_options),
     '#default_value' => $item['weight'],
     '#description' => t('Optional. In the menu, the heavier links will sink and the lighter links will be positioned nearer the top.'),
+    '#prefix' => '<div id="menu-parent-weight-wrapper">',
+    '#suffix' => '</div>',
```

Comment:
Why not wrap this in a `'#type' => 'container'` and use `'#attributes'` for the HTML ID? `'#prefix'`/`'#suffix'` should be reserved for actual (visible) pre-/suffixes where possible rather than HTML, plus adding the wrapper via the FAPI ensures that it will inherit any theme customizations and core improvements in the future (a11y or otherwise), whereas having it added here manually means that you also have to manually account for that later.

Reply:
The existing code uses

```
    '#prefix' => '<div id="menu-parent-select-wrapper">',
    '#suffix' => '</div>',
```

for the `parent` element, so I'm adding the id to the `weight` element the same way for parallelism/consistency. Do you think both should be changed?

Reply:
Yes please 👍🏼 ...I know that there are other instances in core where this is done, but we don't need to propagate bad (or less-than-ideal) practices that we may have inherited from D7 or older versions of Drupal. We need to be adding HTML and attributes in a consistent way that can be extended/improved. So what I said for the previous FAPI element applies here too.

I think that we can use `'#prefix'`/`'#suffix'` in such an "unorthodox" way if:
- there are technical issues and the "proper" way doesn't work or causes problems
- any other instance where we are trying to prepend or append something that cannot be done in any other, more appropriate way.

Reply:
I've made the change to both elements. 

I note that a search through core code for `'#prefix' => '<div id=` turns up 24 occurrences, most of which look to be similarly unorthodox. (So maybe a separate issue is needed?)

Reply:
> and use '#attributes' for the HTML ID?

Or `#id`, which what I used here.

Reply:
One potential issue in replacing the existing `'#prefix' => '<div id=...` with a `'#type' => 'container'` wrapper; that changes the structure of `$form`, so any client that used `hook_form_alter()` to further modify the `$form['parent']` element of the form won't find it where it used to be. Is that serious enough to warrant keeping this element the way it currently is?

Reply:
@bugfolder if all we really need is an HTML id to target with AJAX, then we should try to do that with as minimal a change as possible, yet lets please use something more "standardized" rather than using `'#prefix'` to add HTML to the output, which feels like a hack (the HTML we are prepending cannot be altered via standard hooks/alters etc.).

> I note that a search through core code for `'#prefix' => '<div id=` turns up 24 occurrences, most of which look to be similarly unorthodox. (So maybe a separate issue is needed?)

Yes, I know that this might have been done a lot in the past (things we've inherited from D7 too), but it doesn't mean that we should continue the tradition if it is bad practice. I'm not saying never do it - only do it as a last resort when there aren't any standardized alternatives. ...and yes, we should consider fixing those 24 "nasties" of the past too 👍🏼 

If adding a `'#type' => 'container'` is too "drastic", then how about [`#wrapper_attributes`](https://docs.backdropcms.org/form_api#wrapper_attributes) instead? (which do work with `'#type' => 'select'`)

Reply:
Per discussion in the issue, I've preserved the existing HTML structure for the existing form item (for the benefit of B/C w/ themers) but am using `#wrapper_attributes` for new stuff.